### ISSUE TITLE
Add User Objects

### ISSUE THREAD

**Issue by kaendfinger** [DirectMyFile/irc.dart#1]
In order to improve the API, it would be nice to include user objects.

- [x] Implement WHOIS Support
- [x] Create User Objects
- [x] Add User Tracking

```dart
class User {
  String nickname;
  String username;
  List<Channel> channels = [];
  String mode;
}
```

**Comment by logangorence** [comment]
@kaendfinger Mind doing a thorough review over the feature_user_objects branch?

**Comment by logangorence** [comment]
Closed from c20ec4cb955ce26d1330d1601012c6e3a28593bf.
